src/main/java/it/gov/pagopa/pu/migration/wf/service/ingestion/debtposition/DebtPositionProcessingService.java

```diff
@@ -0,0 +1,194 @@
+  }
+
+  private void handleReaderExceptions(Path file, List<CsvException> readerException, List<DebtPositionErrorDTO> errorList, List<String> unsuccessfulParsedFiles) {
+    if (readerException != null && !readerException.isEmpty()) {
+      for (Exception ex : readerException) {
+        DebtPositionErrorDTO error = new DebtPositionErrorDTO();
+        error.setFileName(file.getFileName().toString());
+        error.setErrorMessage(ex.getMessage());
+        errorList.add(error);
+        unsuccessfulParsedFiles.add(file.getFileName() + ":" + ex.getMessage());
+      }
+      log.error("Error reading file {}: {}", file.getFileName(), readerException);
+    }
+  }
+
+  private static String buildErrorDescription(List<String> unsuccessfulParsedFiles) {
+    String errorDescription = null;
+    if (!unsuccessfulParsedFiles.isEmpty()) {
+      errorDescription = "There were some errors during Debt positions migration.";
+      errorDescription += "\n" + String.join("\n", unsuccessfulParsedFiles);
+    }
+    return errorDescription;
+  }
+
+  protected DebtPositionErrorDTO buildErrorDto(String fileName, long lineNumber, String errorCode, String message) {
+    return DebtPositionErrorDTO.builder()
+      .fileName(fileName)
+      .rowNumber(lineNumber)
+      .errorCode(errorCode)
+      .errorMessage(message)
+      .build();
+  }
+
+
+    public DebtPositionMigrationFileResult processMultipleDebtPositionFiles(List<Path> retrievedFiles, Uploads upload, List<DebtPositionErrorDTO> errorList) {
+        if (retrievedFiles == null || retrievedFiles.isEmpty()) {
+            return DebtPositionMigrationFileResult.builder()
+                .parsedFiles(new ArrayList<>())
+                .numCorrectlyProcessedFiles(0)
+                .numTotalFiles(0)
+                .numTotalRows(0L)
+                .numCorrectlyProcessedRows(0L)
+                .errorDescription(null)
+                .build();
+        }
+        List<Path> parsedFiles = new ArrayList<>();
+        List<String> unsuccessfulParsedFiles = new ArrayList<>();
+        long totalRows = 0;
+        long totalProcessedRows = 0;
+        int numTotalFiles = 0;
+        int numCorrectlyProcessedFiles = 0;
+
+        for (Path file : retrievedFiles) {
+            List<CsvException> readerException = new ArrayList<>();
+            List<InstallmentIngestionFlowFileDTO> dtos = parseCsvFile(file, errorList, unsuccessfulParsedFiles);
+            Iterator<InstallmentIngestionFlowFileDTO> iterator = dtos.iterator();
+            DebtPositionMigrationFileResult singleResult = processDebtPositionFile(
+                iterator,
+                readerException,
+                upload,
+                errorList,
+                file.getParent(),
+                file.getFileName().toString()
+            );
+            String originalFileName = file.getFileName().toString();
+            String tempFileName = "parsed-" + originalFileName.replaceFirst("\\.[^.]+$", "") + ".csv";
```

Comment:
you should provide the version which the workflow-worker is supposed to read the file (2_0)